explorer/testdata/class/fail_instantiate_abstract_class_constructor.carbon

```diff
@@ -0,0 +1,22 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+// RUN: %{not} %{explorer-run}
+// RUN: %{not} %{explorer-run-trace}
+
+package ExplorerTest api;
+
+abstract class C {
+  fn Create() -> Self {
+    return { .a = 1 };
```

Comment:
Hmmm... should this line be an error, because the struct can't convert to an instance of `Self`?

Put differently, can you write the code:

```
fn Foo(c: C) {}

Foo(C.Create());
```

I think that may not hit the variable definition code you've added.

Reply:
And to answer your question, yes, `Foo(C.Create());` _does_ work :). Given this could be parametrized and not resolved yet, went the TODO route.